### PR TITLE
Unit Tests for GQL Backend | Closes #2589

### LINKED ISSUE #2589: Add Unit Tests for GQL Backend
- PHP tests recreate the folder path for the tested file within the `{plugin_root}/tests/` folder. So for example, the tests for `core/domain/services/graphql/` should go into `/tests/testcases/core/domain/services/graphql/`

- if the classes being tested require logic from EE or WP core to operate (such as actions and filters), then they should extend `EE_UnitTestCase` which can be found in the `/tests/includes/` folder. It includes a lot of boilerplate to help simulate normal plugin operation.

- if your class being tested is sufficiently decoupled from other systems, then plz feel free to have your test cases extend `PHPUnit_Framework_TestCase` directly, which will be far more performant.

more information about testing can be found here:
https://github.com/eventespresso/event-espresso-core/blob/master/docs/B--Automated-Testing/automated-testing-in-event-espresso.md

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR adds tests for GQL backend.

### CODE REVIEW

File: tests/testcases/core/domain/services/graphql/connections/PriceConnectionQueriesTest.php [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    public function setUp()
+    {
+        parent::setUp();
+        if (PHP_VERSION_ID < 70000) {
+            $this->markTestSkipped(
+                'WP GraphQL compatible with PHP 7+ only'
+            );
+            return;
+        }
+
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/connections/PriceConnectionQueriesTest.php:25]: doesn't need to change, but the `$subscriber` as well as the test skipping could have been added to `GraphQLUnitTestCase` since it appears to be used in every test case class

File: tests/testcases/core/domain/services/graphql/connections/PriceConnectionQueriesTest.php [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        $this->app_context = new \WPGraphQL\AppContext();
+
+        $this->app_info = new \GraphQL\Type\Definition\ResolveInfo(array());
+    }
+
+    public function tearDown()
+    {
+        parent::tearDown();
+    }
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/connections/PriceConnectionQueriesTest.php:42]: unnecessary since all it does is call the same method on the parent

File: tests/testcases/core/domain/services/graphql/connections/PriceTypeConnectionQueriesTest.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    {
+
+        // Create entities
+        $created_entities = [];
+        for ($i = 1; $i <= $count; $i ++) {
+            $created_entities[ $i ] = $this->new_model_obj_with_dependencies('Price_Type');
+        }
+
+        return $created_entities;
+    }
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/connections/PriceTypeConnectionQueriesTest.php:59]: another good candidate for parent class logic, but with the model name passed as a parameter:

```suggestion
    public function create_entities($model_name, $count = 20)
    {
        // Create entities
        $created_entities = [];
        for ($i = 1; $i <= $count; $i ++) {
            $created_entities[ $i ] = $this->new_model_obj_with_dependencies($model_name);
        }
        return $created_entities;
    }
```

File: tests/testcases/core/domain/services/graphql/mutators/DatetimeDeleteTest.php [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+        /**
+         * Execute the request
+         */
+        $result = $this->deleteMutation($guid);
+
+        /**
+         * The deletion should fail because we're a subscriber
+         */
+        $this->assertArrayHasKey('errors', $result);
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/mutators/DatetimeDeleteTest.php:89]: nice 🥇

File: tests/testcases/core/domain/services/graphql/mutators/DatetimeDeleteTest.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+        /**
+         * Try to delete one more time, and now there's nothing to delete, not even from the trash
+         */
+        $result = $this->deleteMutation($guid, true);
+
+        /**
+         * Now we should have errors again, because there's nothing to be deleted
+         */
+        $this->assertArrayHasKey('errors', $result);
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/mutators/DatetimeDeleteTest.php:136]: NOICE 👌 

good job testing alternate/edge cases

File: tests/testcases/core/domain/services/graphql/mutators/DatetimeUpdateTest.php [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+use GraphQLRelay\Relay;
+
+class DatetimeUpdateTest extends GraphQLUnitTestCase
+{
+    public $name;
+    public $description;
+    public $client_mutation_id;
+    public $admin;
+    public $subscriber;
+    public $entity;
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/mutators/DatetimeUpdateTest.php:15]: ya totally could have saved yourself a bunch of copy n pasting by putting these on the parent class

File: tests/testcases/core/domain/services/graphql/mutators/PriceDeleteTest.php [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+        $variables = wp_json_encode([
+            'input' => [
+                'clientMutationId'  => $this->client_mutation_id,
+                'id'                => $id,
+                'deletePermanently' => $deletePermanently,
+            ]
+        ]);
+
+        return do_graphql_request($mutation, 'deleteEntity', $variables);
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/mutators/PriceDeleteTest.php:69]: even this logic appears to be copied a few times and could have been moved to the parent class, so then the text case classes would only have to pass their specific details to the parent method:

```php
        $mutation = '
		mutation DeletePrice($input: DeleteEspressoPriceInput!) {
            deleteEspressoPrice(input: $input) {
				clientMutationId
				espressoPrice {
                    id
                    dbId
					name
                    isTrashed
				}
			}
		}
		';
        return $this->runDeleteMutation($mutation, $id, $deletePermanently);
```

File: tests/testcases/core/domain/services/graphql/connections/PriceTypeConnectionQueriesTest.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+use EE_Dependency_Map;
+use EE_Error;
+use EEM_Price_Type;
+
+class PriceTypeConnectionQueriesTest extends BaseQueriesTest
+{
+    public function setUp()
+    {
+        $this->model_name = 'Price_Type';
+        $this->skip_create_entities = true;
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/connections/PriceTypeConnectionQueriesTest.php:14]: 👍

File: tests/testcases/core/domain/services/graphql/mutators/BaseMutationTest.php [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    public function runCreateMutation($mutation, $input)
+    {
+        $input['clientMutationId'] = $this->client_mutation_id;
+
+        $variables = wp_json_encode([
+            'input' => $input
+        ]);
+
+        return do_graphql_request($mutation, 'createEntity', $variables);
+    }
```
Comment (tn3rb) [inline tests/testcases/core/domain/services/graphql/mutators/BaseMutationTest.php:54]: awesome 💯
